crates/telemetry/src/lib.rs

```diff
@@ -16,9 +17,35 @@ pub use propagation::inject_trace_context;
 /// Initializes telemetry for Spin using the [tracing] library.
 ///
 /// Under the hood this involves initializing a [tracing::Subscriber] with multiple [Layer]s. One
-/// [Layer] emits [tracing] events to stderr, and another sends spans to an OTEL collector.
+/// [Layer] emits [tracing] events to stderr, another sends spans to an OTel collector, and another
+/// sends metrics to an OTel collector.
 ///
-/// Configuration is pulled from the environment.
+/// Configuration for the OTel layers is pulled from the environment.
+///
+/// Examples of emitting traces from Spin:
+///
+/// ```no_run
+/// #[instrument(name = "span_name", Err(level = Level::INFO), fields(otel.name = "dynamically set name"))]
```

Comment:
Unless I'm mistaken...

```suggestion
/// #[instrument(name = "span_name", err(level = Level::INFO), fields(otel.name = "dynamically set name"))]
```

Reply:
Also, maybe leave off `level`? We probably _typically_ want errors emitted at error level.

```suggestion
/// #[instrument(name = "span_name", err, fields(otel.name = "dynamically set name"))]
```

Reply:
FWIW I've been doing `level = Level::INFO` throughout the whole codebase already. My reasoning has been if we let it emit the event at the default ERROR level then it ends up getting logged to the `fmt_layer` which I figured we don't want. Just b/c some function deep in the guts of Spin is failing doesn't mean we want to log an error to the console.

If you think we should stop following this pattern then I'm happy to change the recommendation here.

Reply:
I guess it depends on the function 🤷